### PR TITLE
ANDROID-14825 Add fonts in Mistica Catalog app

### PR DESCRIPTION
**Pull request by pmartinbTEF**
### :goal_net: What's the goal?
Add fonts in Mistica Catalog app

### :construction: How do we do it?
Added font archives and created correspondent themes for xml and Brand objects for Compose

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, tag the PR with "Breaking Change" label and remember to include breaking change migration guide in release notes where this version is released.
- [x] Tested with dark mode.
- [x] Tested with API 24.
- [x] Sync done with iOS team for this feature to ensure alignment, if applies.
- [x] Accessibility considerations.

### :test_tube: How can I test this?
| v14.1.0 With fonts        | v15.0.1 With fonts           |
| ------------- |-------------|
| drawing      | drawing |
| drawing      | drawing |

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt [hunk 1]
```diff
@@ -46,14 +46,14 @@ class CatalogMainActivity : AppCompatActivity() {
...
-            ComponentStyle("O2", R.style.MisticaTheme_O2, O2),
-            ComponentStyle("Vivo", R.style.MisticaTheme_Vivo, VIVO),
-            ComponentStyle("Vivo New", R.style.MisticaTheme_VivoNew, VIVO_NEW),
-            ComponentStyle("Telefonica", R.style.MisticaTheme_Telefonica, TELEFONICA),
+            ComponentStyle("Movistar", R.style.CatalogMovistar, MOVISTAR),
+            ComponentStyle("O2", R.style.CatalogO2, O2),
+            ComponentStyle("Vivo", R.style.CatalogVivo, VIVO),
+            ComponentStyle("Vivo New", R.style.CatalogVivoNew, VIVO_NEW),
+            ComponentStyle("Telefonica", R.style.CatalogTelefonica, TELEFONICA),
             ComponentStyle("Blau", R.style.MisticaTheme_Blau, BLAU),
```
Comment (pmartinbTEF) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:54]: Blau has a licensed font

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+import com.telefonica.mistica.compose.theme.brand.TuBrand
+import com.telefonica.mistica.compose.theme.brand.VivoBrand
+import com.telefonica.mistica.compose.theme.brand.VivoNewBrand
+import com.telefonica.mistica.compose.theme.brushes.MisticaBrushes
+import com.telefonica.mistica.compose.theme.color.MisticaColors
+import com.telefonica.mistica.compose.theme.values.MisticaRadius
+
+object CatalogMovistarBrand : Brand {
+    override val compatibilityTheme: Int
+        get() = R.style.MisticaTheme_Movistar
```
Comment (yamal-alm) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:22]: this should be `CatalogMovistar` so `AndroidViews` still use the correct font

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+import com.telefonica.mistica.compose.theme.brand.O2NewBrand
+import com.telefonica.mistica.compose.theme.brand.TelefonicaBrand
+import com.telefonica.mistica.compose.theme.brand.TuBrand
+import com.telefonica.mistica.compose.theme.brand.VivoBrand
+import com.telefonica.mistica.compose.theme.brand.VivoNewBrand
+import com.telefonica.mistica.compose.theme.brushes.MisticaBrushes
+import com.telefonica.mistica.compose.theme.color.MisticaColors
+import com.telefonica.mistica.compose.theme.values.MisticaRadius
+
+object CatalogMovistarBrand : Brand {
```
Comment (yamal-alm) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:20]: we can create a `CatalogBrand` and reuse it so we don't have to write all this code for every brand:
```kotlin
abstract class CatalogBrand(
     private val basebBrand: Brand,
     val theme: Int,
) : Brand {
     override val compatibilityTheme: Int = theme
     override val lightColors: MisticaColors = brand.lightColors
     // ...
}
```

The you can use it as follows:
```kotlin
object CatalogMovistarBrand : CatalogBrand(MovistarBrand, R.style.CatalogMovistar) {
    override val fontFamily: FontFamily
        get() = FontFamily(
            Font(R.font.onair_light, FontWeight.Light),
            Font(R.font.onair_regular, FontWeight.Normal),
            Font(R.font.onair_medium, FontWeight.Medium),
        )
}
```

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt (hunk 1 above)
Comment (dagonco) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:54]: Even if we dont have that font in our catalog could we have a CatalogBlau like the rest of the brands, even if is empty?

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt (hunk 3 above)
Comment (pmartinbTEF) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:20]: Done!

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt (hunk 1 above)
Comment (pmartinbTEF) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:54]: Done!

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt (hunk 2 above)
Comment (pmartinbTEF) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt]: Done

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+object CatalogMovistarBrand : CatalogBrand(MovistarBrand, R.style.CatalogMovistar) {
+    override val fontFamily: FontFamily
+        get() = FontFamily(
+            Font(R.font.onair_light, FontWeight.Light),
+            Font(R.font.onair_regular, FontWeight.Normal),
+            Font(R.font.onair_medium, FontWeight.Medium),
+        )
+}
+
+object CatalogO2Brand : CatalogBrand(O2Brand, R.style.CatalogO2) {
```
Comment (dagonco) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:59]: A minor, could we have O2 with his brother O2New together as Vivo is with its two themes?

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+import com.telefonica.mistica.compose.theme.brushes.MisticaBrushes
+import com.telefonica.mistica.compose.theme.color.MisticaColors
+import com.telefonica.mistica.compose.theme.values.MisticaRadius
+
+abstract class CatalogBrand(
+    baseBrand: Brand,
+    val theme: Int,
+) : Brand {
+    override val compatibilityTheme: Int
+        get() = theme
```
Comment (dagonco) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:25]: Now seeing this, it is necessary that theme is computed every time that compatibilityTheme is retrieved? cc: @yamal-alm

**Review by dagonco — APPROVED** [review]
Cool, nice job!

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt (hunk 5 above)
Comment (yamal-alm) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:25]: yes, we can use `override val compatibilityTheme: Int = theme` instead

**Review by yamal-alm — APPROVED** [review]
good job 👏

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt (hunk 4 above)
Comment (pmartinbTEF) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:59]: Done!

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt (hunk 5 above)
Comment (pmartinbTEF) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/common/CatalogBrands.kt:25]: Done

**Review by aweell — APPROVED** [review]
LGTM

**Pull request merged**